OpenEdXMobile/src/main/java/org/edx/mobile/module/registration/view/RegistrationEditTextView.java

```diff
@@ -92,13 +93,23 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
                     isChangedByUser = true;
                     return;
                 }
-                isValidInput();
+                // Don't show the error until view has lost the focus at least once
+                if (hasFocusLost) {
+                    isValidInput();
+                }
             }
 
             @Override
             public void afterTextChanged(Editable s) {
             }
         });
+
+        mEditText.setOnFocusChangeListener((v, hasFocus) -> {
+            if (!hasFocus) {
+                isValidInput();
+                hasFocusLost = true;
+            }
+        });
     }
```

Comment:
I think the following lines of code will cover the story requirement
```java
        // Add text change listener
        mEditText.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                // Clear Error from the field
                if (mErrorTextView.getVisibility() == View.VISIBLE) {
                    mErrorTextView.setText(null);
                    mErrorTextView.setVisibility(View.GONE);
                    mTextInputLayout.setContentDescription(null);
                }
            }

            @Override
            public void afterTextChanged(Editable s) {
            }
        });

        mEditText.setOnFocusChangeListener((v, hasFocus) -> {
            if (!hasFocus) {
                isValidInput();
            }
        });
```

Reply:
As per discussion on [LEARNER-7767](https://openedx.atlassian.net/browse/LEARNER-7767) *Hilary* approved the iOS follow suggested by @saeedbashir